src/database/DataTypes/RequisitionItem.js

```diff
@@ -230,6 +229,60 @@ export class RequisitionItem extends Realm.Object {
   get lastRequisitionDate() {
     return this.item?.lastRequisitionDate;
   }
+
+  setIncomingStock(value) {
+    this.incomingStock = value;
+    this.recalculateStockOnHand();
```

Comment:
For each of the values which contribute to the stock on hand/closing balance, call `recalculateStockOnHand` after setting the value. You can technically directly set things i.e:

```
UIDatabase.write(() => {
    someRequisitionItem.positiveAdjustments = 40
})
```

which will put the `stockOnHand` out of wack, which isn't great!

Reply:
If you create a setter `positiveAdjustments`, will it override accessing the schema value directly or just break things? Another option is using listeners (i.e. db trigger approach) but... ew haha.

Reply:
I don't remember the exact behaviour, but it breaks things: the schema would/should be `_positiveAdjustments` (or variant), with the get/sets being `get positiveAdjustments`/`set positiveAdjustments`, which essentially has the same problem (This was used for `Report._data`, for example`